src/_cffi_src/openssl/dsa.py

```diff
@@ -38,9 +38,81 @@
 """
 
 MACROS = """
+/* DSAparams_dup is a macro in 0.9.8 */
+DSA *DSAparams_dup(DSA *);
 int DSA_generate_parameters_ex(DSA *, int, unsigned char *, int,
                                int *, unsigned long *, BN_GENCB *);
+/* added in 1.1.0 to access the opaque struct */
+void DSA_get0_pqg(const DSA *, BIGNUM **, BIGNUM **, BIGNUM **);
+int DSA_set0_pqg(DSA *, BIGNUM *, BIGNUM *, BIGNUM *);
+void DSA_get0_key(const DSA *, BIGNUM **, BIGNUM **);
+int DSA_set0_key(DSA *, BIGNUM *, BIGNUM *);
```

Comment:
:+1:

On Sat, Apr 30, 2016 at 9:31 PM, Paul Kehrer <notifications@github.com>
wrote:

> In src/_cffi_src/openssl/dsa.py
> <https://github.com/pyca/cryptography/pull/2889#discussion_r61677892>:
>
> >  int DSA_generate_parameters_ex(DSA *, int, unsigned char *, int,
> >                                 int *, unsigned long *, BN_GENCB *);
> > +/* added in 1.1.0 to access the opaque struct */
> > +void DSA_get0_pqg(const DSA *, BIGNUM **, BIGNUM **, BIGNUM **);
> > +int DSA_set0_pqg(DSA *, BIGNUM *, BIGNUM *, BIGNUM *);
> > +void DSA_get0_key(const DSA *, BIGNUM **, BIGNUM **);
> > +int DSA_set0_key(DSA *, BIGNUM *, BIGNUM *);
>
> Ah, good point. Since they're functions (either available in 1.1.0 or
> declared by us) we can put them in FUNCTIONS unconditionally. This same
> issue is present in the RSA PR.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly or view it on GitHub
> <https://github.com/pyca/cryptography/pull/2889/files/6fc6c492423842d038e30fd67b0b504730c27d9a#r61677892>
>



-- 
"I disapprove of what you say, but I will defend to the death your right to
say it." -- Evelyn Beatrice Hall (summarizing Voltaire)
"The people's good is the highest law." -- Cicero
GPG Key fingerprint: D1B3 ADC0 E023 8CA6